File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.js
/**
 * @param {number} target
 * @param {number[]} nums
 * @return {number}
 */
const minSubArrayLen = function(target, nums) {
    let leftIdx = 0;
    let minLen = Infinity;
    let sum = 0;
    
    for (let rightIdx = 0; rightIdx < nums.length; rightIdx++) {
        sum += nums[rightIdx];

        while (sum >= target) {
            minLen = Math.min(minLen, rightIdx - leftIdx + 1);
            sum -= nums[leftIdx];
            leftIdx++;
        }
    }

    return minLen === Infinity ? 0 : minLen;
};

File: 0209-minimum-size-subarray-sum/0209-minimum-size-subarray-sum.py
class Solution(object):
    def minSubArrayLen(self, target, nums):
        """
        :type target: int
        :type nums: List[int]
        :rtype: int
        """
        left, right, total, N = 0, 0, 0, len(nums)
        min_len = float('inf')

        while right < N:
            total += nums[right]
            while total >= target:
                min_len = min(min_len, right-left+1)
                total -= nums[left]
                left += 1
            right += 1
        return min_len if min_len != float('inf') else 0